s2cloudless/sentinelhub_masking.py

```diff
@@ -201,44 +205,52 @@ def _download_bands_and_valid_data_mask(self):
         norm_factors = [response["userdata.json"]["norm_factor"] for response in responses]
         del responses
 
+        if data is None:
+            raise ValueError()
+
         self.bands = data[..., :-1]
         self.data_mask = data[..., -1] != 0
 
         normalized_bands = (np.round(array * factor, 4) for array, factor in zip(self.bands, norm_factors))
         self.bands = np.asarray(list(normalized_bands), dtype=np.float32)
 
-    def get_probability_masks(self, non_valid_value=0):
+    def get_probability_masks(self, non_valid_value: int = 0) -> np.ndarray:
         """
         Get probability maps of areas for each available date. The pixels without valid data are assigned
         non_valid_value.
 
         :param non_valid_value: Value to be assigned to non valid data pixels
-        :type non_valid_value: float
         :return: Probability map of shape `(times, height, width)` and `dtype=numpy.float64`
-        :rtype: numpy.ndarray
         """
         # pylint: disable=invalid-unary-operand-type
+
         if self.probability_masks is None:
             self.get_data()
+            if self.bands is None:
+                raise ValueError("Bands should not be None.")
+
             self.probability_masks = self.cloud_detector.get_cloud_probability_maps(self.bands)
 
+        if self.data_mask is None:
+            raise ValueError("Data mask should not be None.")
+
         self.probability_masks[~self.data_mask] = non_valid_value
         return self.probability_masks
 
-    def get_cloud_masks(self, threshold=None, non_valid_value=0):
+    def get_cloud_masks(self, threshold: Optional[float] = None, non_valid_value: int = 0) -> np.ndarray:
         """The binary cloud mask is computed on the fly. Be cautious. The pixels without valid data are assigned
         non_valid_value.
 
         :param threshold: A float from [0,1] specifying threshold
-        :type threshold: float
         :param non_valid_value: Value which will be assigned to pixels without valid data
-        :type non_valid_value: int in range `[-254, 255]`
         :return: Binary cloud masks of shape `(times, height, width)` and `dtype=numpy.int8`
-        :rtype: numpy.ndarray
         """
         # pylint: disable=invalid-unary-operand-type
         self.get_probability_masks()
 
+        if self.probability_masks is None or self.data_mask is None:
+            raise ValueError("Both data mask and probability masks should not be None.")
```

Comment:
the first line returns a `np.ndarray`, can you just pass that forward to the cloud detector :thinking: 

is the error then still correct?

Reply:
both updated